src/mongocrypt-ctx.c

```diff
@@ -294,14 +293,9 @@ mongocrypt_ctx_t *mongocrypt_ctx_new(mongocrypt_t *crypt) {
         CLIENT_ERR("cannot create context from uninitialized crypt");
         return NULL;
     }
-    ctx_size = sizeof(_mongocrypt_ctx_encrypt_t);
-    if (sizeof(_mongocrypt_ctx_decrypt_t) > ctx_size) {
-        ctx_size = sizeof(_mongocrypt_ctx_decrypt_t);
-    }
-    if (sizeof(_mongocrypt_ctx_datakey_t) > ctx_size) {
-        ctx_size = sizeof(_mongocrypt_ctx_datakey_t);
-    }
-    ctx = bson_malloc0(ctx_size);
+
+    // Allocate with memory and alignment large enough for any possible context type.
+    ctx = bson_aligned_alloc0(MONGOCRYPT_CTX_ALLOC_ALIGNMENT, MONGOCRYPT_CTX_ALLOC_SIZE);
```

Comment:
Inlining these macros into an expression _might_ lead to recomputation of the constants (`max(max(...))`) per invocation of this function... or it might not (premature optimization?). I'll leave the decision to revert to using `static const` or not up to you.